src/App.tsx

```diff
@@ -1,14 +1,62 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+
+enum Status {
+  Active = 'active',
+  Completed = 'completed',
+}
+
+function searchInTitle(title: string, search: string) {
+  return title.toLowerCase().includes(search.trim().toLowerCase());
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todo, setTodo] = useState<Todo | null>(null);
+  const [filterByStatus, setFilterByStatus] = useState('all');
+  const [query, setQuery] = useState('');
+
+  const onTodo = (selectedTodo: Todo | null) => {
+    setTodo(selectedTodo);
+  };
+
+  let selectedTodos;
+  let visibleTodos;
+
+  switch (filterByStatus) {
+    case Status.Active:
+      selectedTodos = todos.filter(currentTodo => !currentTodo.completed);
+      break;
+
+    case Status.Completed:
+      selectedTodos = todos.filter(currentTodo => currentTodo.completed);
+      break;
+
+    default:
+      selectedTodos = todos;
+      break;
+  }
```

Comment:
I would go for something like this:
```suggestion
  const filteredTodos = () => {
    switch (filterByStatus) {
    case Status.Active:
      return todos.filter(currentTodo => !currentTodo.completed);

    case Status.Completed:
      return todos.filter(currentTodo => currentTodo.completed);

    default:
      return todos;
  }
  };
```

And similar thing for `visibleTodos` ;)

Reply:
And then in JSX you just do `filteredTodos().map`

Reply:
what an oversight, thanks!

Reply:
I don't know exactly why do you want me to map it. I hope my refactor is sufficient